plugin/arcconf/arcconf.py

```diff
@@ -0,0 +1,963 @@
+            (cur_ctrl_num, disk_channel, disk_device, disk_status) = cur_disk.plugin_data.split(',')
+            if ctrl_num and cur_ctrl_num != ctrl_num:
+                raise LsmError(ErrorNumber.INVALID_ARGUMENT, "Illegal input disks argument: disks are not from the same controller/system.")
+            if vol_created > 0:
+                break
+
+            ctrl_num = cur_ctrl_num
+            arcconf_raid_level = 'simple_volume'
+            disk_channel = str(disk_channel.strip())
+            disk_device = str(disk_device.strip())
+            vol_size = _lsm_size_to_arcconf_MB(size)
+            
+            try:
+                if int(disk_status) == Disk.STATUS_OK:
+                    cmds = ['create', str(ctrl_num), 'logicaldrive', str(vol_size), arcconf_raid_level, disk_channel, disk_device]
+                    try:
+                        if vol_created == 0:
+                            self._arcconf_exec( cmds, COMPONENT[5], flag_convert=False, flag_force=True)
+                            vol_created = 1
+                            vol_id = str(vol_id + 1)
+                    except:
+                        continue
+                else:
+                    continue
+            except ExecError:
+            # Check whether disk is free
+                requested_disk_ids = [d.id for d in self.disks()]
+            
+                if cur_disk.id in requested_disk_ids and \
+                    not cur_disk.status & Disk.STATUS_FREE:
+                    raise LsmError(ErrorNumber.DISK_NOT_FREE, "Disk %s is not in STATUS_FREE state" % cur_disk.id)
+            
+            
+        pool_id  = str(args.id)
+        vol_list = self.volumes()
+        vol_last = vol_list[-1]
+        vol_id = vol_last.id
+        lsm_vols = self.volumes(search_key='id', search_value=vol_id)
+        
+        if len(lsm_vols) == 0:
+            raise LsmError(
+                ErrorNumber.PLUGIN_BUG,
+                "volume_create(): Got unexpected count(not > 0) of new "
+                "volumes: %s" % lsm_vols)
+
+        return vol_id, lsm_vols[-1]
+        
+
+    @_handle_errors
+    def volume_delete(self, volume, flags=0):
+        """
+        Depends on command:
+            arcconf delete <ctrlNo> logicaldrive <ld#> noprompt
+        """
+        if not volume.plugin_data:
+            raise LsmError(
+                ErrorNumber.INVALID_ARGUMENT,
+                "Illegal input volume argument: missing plugin_data property")
+
+        (ctrl_num, array_num, ld_num) = volume.plugin_data.split(":")
+
+        try:
+            self._arcconf_exec(['delete', ctrl_num, 'logicaldrive', array_num], COMPONENT[5], flag_convert=False, flag_force=True)
+        except ExecError:
+            raise LsmError(
+                ErrorNumber.NOT_FOUND_VOLUME,
+                "Volume not found")
+
+        return None
+
+    def volume_enable(self, volume, flags=Client.FLAG_RSVD):
```

Comment:
Again for this chunk, remove if not supported and update Capabilities accordingly.